### PR TITLE
[CARBONDATA-1123] Rename interface and variable for RLE encoding

### PR DESCRIPTION
**Pull request by jackylk**
This PR improve the readability in RLE encoding implementation. No logic is changed in this PR.
Basically, RLE and inverted index will output 4 kinds of data:
1. dataPage for the dimension data
2. dataRlePage for the RLE encoded dimension data
3. rowIdPage for the rowId
4. rowIdRlePage for the RLE encoded RowId data

Verify with all testsuites

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/2217/

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/carbondata-pr-spark-1.6/87/

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/carbondata-pr-spark-1.6/325/<h2>Build result: FAILURE</span></h2>[...truncated 60.75 KB...][ERROR] Re-run Maven using the -X switch to enable full debug logging.[ERROR] [ERROR] For more information about the errors and possible solutions, please read the following articles:[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException[ERROR] [ERROR] After correcting the problems, you can resume the build with the command[ERROR]   mvn <goals> -rf :carbondata-processing[JENKINS] Archiving /home/jenkins/jenkins-slave/workspace/carbondata-pr-spark-1.6/processing/pom.xml to org.apache.carbondata/carbondata-processing/1.2.0-SNAPSHOT/carbondata-processing-1.2.0-SNAPSHOT.pom[JENKINS] Archiving /home/jenkins/jenkins-slave/workspace/carbondata-pr-spark-1.6/integration/spark-common/pom.xml to org.apache.carbondata/carbondata-spark-common/1.2.0-SNAPSHOT/carbondata-spark-common-1.2.0-SNAPSHOT.pom[JENKINS] Archiving /home/jenkins/jenkins-slave/workspace/carbondata-pr-spark-1.6/examples/spark/pom.xml to org.apache.carbondata/carbondata-examples-spark/1.2.0-SNAPSHOT/carbondata-examples-spark-1.2.0-SNAPSHOT.pom[JENKINS] Archiving /home/jenkins/jenkins-slave/workspace/carbondata-pr-spark-1.6/integration/hive/pom.xml to org.apache.carbondata/carbondata-hive/1.2.0-SNAPSHOT/carbondata-hive-1.2.0-SNAPSHOT.pom[JENKINS] Archiving /home/jenkins/jenkins-slave/workspace/carbondata-pr-spark-1.6/common/pom.xml to org.apache.carbondata/carbondata-common/1.2.0-SNAPSHOT/carbondata-common-1.2.0-SNAPSHOT.pom[JENKINS] Archiving /home/jenkins/jenkins-slave/workspace/carbondata-pr-spark-1.6/assembly/pom.xml to org.apache.carbondata/carbondata-assembly/1.2.0-SNAPSHOT/carbondata-assembly-1.2.0-SNAPSHOT.pom[JENKINS] Archiving /home/jenkins/jenkins-slave/workspace/carbondata-pr-spark-1.6/pom.xml to org.apache.carbondata/carbondata-parent/1.2.0-SNAPSHOT/carbondata-parent-1.2.0-SNAPSHOT.pom[JENKINS] Archiving /home/jenkins/jenkins-slave/workspace/carbondata-pr-spark-1.6/examples/flink/pom.xml to org.apache.carbondata/carbondata-examples-flink/1.2.0-SNAPSHOT/carbondata-examples-flink-1.2.0-SNAPSHOT.pom[JENKINS] Archiving /home/jenkins/jenkins-slave/workspace/carbondata-pr-spark-1.6/core/pom.xml to org.apache.carbondata/carbondata-core/1.2.0-SNAPSHOT/carbondata-core-1.2.0-SNAPSHOT.pom[JENKINS] Archiving /home/jenkins/jenkins-slave/workspace/carbondata-pr-spark-1.6/hadoop/pom.xml to org.apache.carbondata/carbondata-hadoop/1.2.0-SNAPSHOT/carbondata-hadoop-1.2.0-SNAPSHOT.pom[JENKINS] Archiving /home/jenkins/jenkins-slave/workspace/carbondata-pr-spark-1.6/integration/presto/pom.xml to org.apache.carbondata/carbondata-presto/1.2.0-SNAPSHOT/carbondata-presto-1.2.0-SNAPSHOT.pom[JENKINS] Archiving /home/jenkins/jenkins-slave/workspace/carbondata-pr-spark-1.6/integration/spark-common-test/pom.xml to org.apache.carbondata/carbondata-spark-common-test/1.2.0-SNAPSHOT/carbondata-spark-common-test-1.2.0-SNAPSHOT.pom[JENKINS] Archiving /home/jenkins/jenkins-slave/workspace/carbondata-pr-spark-1.6/integration/spark/pom.xml to org.apache.carbondata/carbondata-spark/1.2.0-SNAPSHOT/carbondata-spark-1.2.0-SNAPSHOT.pomchannel stoppedSetting status of 2fd6d7730e5f1b6b11da15b0d4752832cac0bb7d to FAILURE with url https://builds.apache.org/job/carbondata-pr-spark-1.6/325/ and message: 'Tests Failed for Spark1.6 'Using context: Jenkins(Spark1.6): mvn clean test -Pspark-1.6

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/2444/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/2447/

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/carbondata-pr-spark-1.6/328/

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/carbondata-pr-spark-1.6/336/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/2455/

File: core/src/main/java/org/apache/carbondata/core/datastore/columnar/BlockIndexerStorageForNoInvertedIndexForShort.java [hunk 1]
```diff
@@ -38,42 +38,42 @@
 
   public BlockIndexerStorageForNoInvertedIndexForShort(byte[][] keyBlockInput,
```
Comment (QiangCai) [inline core/src/main/java/org/apache/carbondata/core/datastore/columnar/BlockIndexerStorageForNoInvertedIndexForShort.java]: rename parameter keyBlockInput

File: core/src/main/java/org/apache/carbondata/core/datastore/columnar/BlockIndexerStorageForNoInvertedIndex.java [hunk 2]
```diff
@@ -38,26 +37,26 @@
   private byte[] max;
 
   public BlockIndexerStorageForNoInvertedIndex(byte[][] keyBlockInput) {
```
Comment (QiangCai) [inline core/src/main/java/org/apache/carbondata/core/datastore/columnar/BlockIndexerStorageForNoInvertedIndex.java]: rename parameter keyBlockInput

File: core/src/main/java/org/apache/carbondata/core/datastore/columnar/BlockIndexerStorageForShort.java [hunk 3]
```diff
@@ -71,14 +72,14 @@ public BlockIndexerStorageForShort(byte[][] keyBlock, boolean compressData,
     return columnWithIndexs;
   }
 
-  private short[] extractDataAndReturnIndexes(ColumnWithShortIndex[] columnWithIndexs,
+  private short[] extractDataAndReturnRowId(ColumnWithShortIndex[] dataWithRowId,
       byte[][] keyBlock) {
```
Comment (QiangCai) [inline core/src/main/java/org/apache/carbondata/core/datastore/columnar/BlockIndexerStorageForShort.java]: rename parameter keyBlock, and line 58 also need

File: core/src/main/java/org/apache/carbondata/core/datastore/columnar/BlockIndexerStorageForInt.java [hunk 4]
```diff
@@ -27,25 +27,26 @@
...
-  private int[] dataIndexMap;
+  private int[] dataRlePage;
 
   private int totalSize;
 
-  public BlockIndexerStorageForInt(byte[][] keyBlock, boolean compressData, boolean isNoDictionary,
+  public BlockIndexerStorageForInt(byte[][] dataPage, boolean rleOnData, boolean isNoDictionary,
       boolean isSortRequired) {
-    ColumnWithIntIndex[] columnWithIndexs = createColumnWithIndexArray(keyBlock, isNoDictionary);
+    ColumnWithIntIndex[] dataWithRowId = createColumnWithIndexArray(dataPage, isNoDictionary);
```
Comment (QiangCai) [inline core/src/main/java/org/apache/carbondata/core/datastore/columnar/BlockIndexerStorageForInt.java]: suggest to change as following.
ColumnWithIntIndex => ColumnWithRowId
createColumnWithIndexArray => createColumnWithRowId

File: core/src/main/java/org/apache/carbondata/core/datastore/columnar/BlockIndexerStorageForInt.java [hunk 5]
```diff
@@ -27,25 +27,26 @@
...
     if (isSortRequired) {
-      Arrays.sort(columnWithIndexs);
+      Arrays.sort(dataWithRowId);
     }
-    compressMyOwnWay(extractDataAndReturnIndexes(columnWithIndexs, keyBlock));
-    if (compressData) {
-      compressDataMyOwnWay(columnWithIndexs);
+    int[] rowIds = extractDataAndReturnIndexes(dataWithRowId, dataPage);
```
Comment (QiangCai) [inline core/src/main/java/org/apache/carbondata/core/datastore/columnar/BlockIndexerStorageForInt.java]: suggest to change:
extractDataAndReturnIndexes => extractDataAndReturnRowIds

File: core/src/main/java/org/apache/carbondata/core/datastore/columnar/BlockIndexerStorageForInt.java [hunk 6]
```diff
@@ -189,11 +190,11 @@ private void compressDataMyOwnWay(ColumnWithIntIndex[] indexes) {
     }
     map.add(start);
     map.add(counter);
-    this.keyBlock = convertToKeyArray(list);
-    if (indexes.length == keyBlock.length) {
-      dataIndexMap = new int[0];
+    this.dataPage = convertToKeyArray(list);
```
Comment (QiangCai) [inline core/src/main/java/org/apache/carbondata/core/datastore/columnar/BlockIndexerStorageForInt.java]: convertToDataPage

File: core/src/main/java/org/apache/carbondata/core/datastore/columnar/BlockIndexerStorageForShort.java [hunk 7]
```diff
@@ -27,25 +27,26 @@
...
-  private short[] dataIndexMap;
+  private short[] dataRlePage;
 
   private int totalSize;
 
-  public BlockIndexerStorageForShort(byte[][] keyBlock, boolean compressData,
+  public BlockIndexerStorageForShort(byte[][] dataPage, boolean rleOnData,
       boolean isNoDictionary, boolean isSortRequired) {
-    ColumnWithShortIndex[] columnWithIndexs = createColumnWithIndexArray(keyBlock, isNoDictionary);
+    ColumnWithShortIndex[] dataWithRowId = createColumnWithIndexArray(dataPage, isNoDictionary);
```
Comment (QiangCai) [inline core/src/main/java/org/apache/carbondata/core/datastore/columnar/BlockIndexerStorageForShort.java]: ColumnWithShortIndex => ColumnWithShortRowId
createColumnWithIndexArray => createColumnWithRowId

File: core/src/main/java/org/apache/carbondata/core/datastore/columnar/BlockIndexerStorageForShort.java [hunk 8]
```diff
@@ -27,25 +27,26 @@
...
     if (isSortRequired) {
-      Arrays.sort(columnWithIndexs);
+      Arrays.sort(dataWithRowId);
     }
-    compressMyOwnWay(extractDataAndReturnIndexes(columnWithIndexs, keyBlock));
-    if (compressData) {
-      compressDataMyOwnWay(columnWithIndexs);
+    short[] rowIds = extractDataAndReturnRowId(dataWithRowId, dataPage);
```
Comment (QiangCai) [inline core/src/main/java/org/apache/carbondata/core/datastore/columnar/BlockIndexerStorageForShort.java]: extractDataAndReturnRowId => extractDataAndReturnRowIds

File: core/src/main/java/org/apache/carbondata/core/datastore/columnar/BlockIndexerStorageForShort.java [hunk 9]
```diff
@@ -193,13 +194,13 @@ private void compressDataMyOwnWay(ColumnWithShortIndex[] indexes) {
     map.add(counter);
     // if rle is index size is more than 70% then rle wont give any benefit
     // so better to avoid rle index and write data as it is
-    boolean useRle = (((list.size() + map.size()) * 100) / indexes.length) < 70;
+    boolean useRle = (((list.size() + map.size()) * 100) / dataWithRowId.length) < 70;
     if (useRle) {
-      this.keyBlock = convertToKeyArray(list);
-      dataIndexMap = convertToArray(map);
+      this.dataPage = convertToKeyArray(list);
```
Comment (QiangCai) [inline core/src/main/java/org/apache/carbondata/core/datastore/columnar/BlockIndexerStorageForShort.java]: convertToDataPage

**Comment by jackylk** [comment]
@QiangCai I fixed all comment

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/2545/

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/carbondata-pr-spark-1.6/436/

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/carbondata-pr-spark-1.6/437/

**Comment by QiangCai** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/2553/

**Comment by QiangCai** [comment]
retest this please

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/carbondata-pr-spark-1.6/445/

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/carbondata-pr-spark-1.6/447/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/2555/

**Comment by QiangCai** [comment]
LGTM

**Pull request closed without merging**